### PR TITLE
Add the Rogue Mage line to core

### PR DESCRIPTION
**Pull request by Dalas121**
Liberty contains 3 custom outlaws: the Rogue Mage, Shadow Mage, and Shadow Lord (and TDG has the related "Arcane Enforcer"). Different variants of these units are also used in EI and TDG, as well as (possibly) the upcoming TRoW and HttT projects.

With multiple mainline campaigns using these units (and all with different variations/stats, which is my fault), I suggest deciding on stats and moving the Rogue Mage line to core to standardize the units and make them easier for campaigns/UMC to use.

If adding these to core is acceptable, here are some things that need to be handled:
- [ ] Stats Balancing <details><summary>(this is a summary/detail HTML element; click me for details)</summary><br>With the proposed stats I intend these to be fragile mixed fighters, with good damage at both melee and ranged but very low hp. Nightstalks is primarily for thematic reasons; after all, they're called "Shadow" Mages and wear all black. I felt that Liberty's leadership ability didn't really fit the unit's fantasy.<br><br>Liminal, nightstalks, magical, and "astral" all on the same unit is pretty extreme, and some of them might need to be changed. But I do feel the Shadow Mage's and (separately) the Shadow Lord's unusual standing animations need to be reflected in their stats in some way.<br><br>One more vanilla alternative could be to remove magical from the L2 Shadow Mage's melee and give it an alternate damage type instead. Then remove the "astral" special altogether and give the L3 Shadow Mage magical on melee. Damage of both melees would need to be increased.</details>
- [ ] The Rogue Mage sprites are poor, in my opinion.

- [ ] The male Rogue Mage, Shadow Mage, and Shadow Lord are missing several animations (and the female ones aren't incredible either).

- [ ] None of these units have portraits.

- [x] Liberty may need rebalancing. I'm confident EI and TDG aren't issues.

- [ ] UMC likely have their own versions of these units, which may cause conflicts.

- [ ] Copyright.

- [ ] I may have made mistakes/bugs in the PR.

### CODE REVIEW

**Comment by nemaara** [comment]
Things to check for this pr:

Any strings that assume male that may also now use female gendering? 

Check for any spawns or events that modify hitpoints, as I think these have lower HP? 

Did you run wmllint on these? 

Check gender assumptions in Liberty recruiting and strings (S6 and 7). 


Balancing for Liberty is not an issue.

**Comment by doofus-01** [comment]
I will look into this next week.

**Review by knyghtmare — CHANGES_REQUESTED** [review]
- Missing copyrights.csv addition
- Male Shadow Mage has no melee attack animation graphics
- Male Shadow Lord has no melee attack animation graphics
- Female Shadow Mage/Lord seems better animated

Note: this will impact a lot of UMC including my own, but I can compromise.

**Comment by Dalas121** [comment]
I would like to make sure we're not seriously breaking UMC with this.  Do most UMC use the original Liberty stats? If so we might want to consider keeping those stats unchanged, even if they're not my favorite personally.

**Comment by knyghtmare** [comment]
> I would like to make sure we're not seriously breaking UMC with this. Do most UMC use the original Liberty stats? If so we might want to consider keeping those stats unchanged, even if they're not my favorite personally.
> 
> Or if the issue is ID duplication, I'm unfortunately not sure there's much that can be done about that unless we want to rename the unit line.

In my understanding, there are some that just "pull in" the unit-line from Liberty itself, so uses whatever stats that was in there. I know 2 such cases.
There are some that uses the 1.14 Liberty Rogue-Mage line. (they copied the unit files into the addon), so ID clashes there.
I use maybe the ones with stats you have suggested with just the level 1 unit being "Shadow Initiate" and not "Rogue Mage" (for a campaign purpose).
There might be more extra cases in other UMC I have not explored much.

Also, note that you intend for this in 1.19 so it will give most UMC authors time (~2 years) to readjust their addons so plenty of time to remove/adjust/rebalance their works.

File: data/core/units/humans/Outlaw_Shadow_Lord.cfg [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    [standing_anim]
+        start_time=0
+        [frame]
+            image="units/human-outlaws/shadow-lord.png:70,units/human-outlaws/shadow-lord-[1~7].png:70"
+        [/frame]
+    [/standing_anim]
+    {DEFENSE_ANIM "units/human-outlaws/shadow-lord-defend2.png" "units/human-outlaws/shadow-lord-defend1.png" {SOUND_LIST:HUMAN_OLD_HIT} }
+    [attack]
+        name=astral blade
+        icon="icons/sword-astral.png"
```
Comment (doofus-01) [inline data/core/units/humans/Outlaw_Shadow_Lord.cfg:46]: This attack icon wasn't transferred over.

**Review by doofus-01 — CHANGES_REQUESTED** [review]
I've got mixed feelings about this.  
 - On the one hand, it makes sense to have a unit line used in multiple mainline campaigns be core.
 - On the other hand, adding these does increase the work needed to get core artwork completed.  At least they have animations, but as was originally noted, there is still plenty to work on in the sprites, and now we'd have more missing portraits.

I won't stand in the way of adding the Rogue/Shadow Mages/Lords, if that's the consensus.  Even if the sprites need some work, they are unique and interesting.  It wouldn't at all be starting from zero.

Arcane Enforcer is a different story.  It seems like a new unit and the sprite would need a lot of work, so I'm not seeing the benefit for now.  Not every campaign unit needs to be moved to core, especially if it's a new campaign unit.  I'd ask for this to be removed, or at least some further justification given of why we need it.

**Comment by Dalas121** [comment]
> I've got mixed feelings about this. On the one hand/on the other hand

If it's preferable to keep these in separate campaigns, I'm certainly open to that, and I can handle reconciling the stats/animations for them (that's less work for me than getting them into core).  If it's preferable to move them into core so that all campaigns are consistent

> Arcane Enforcer is a different story. It seems like a new unit and the sprite would need a lot of work

In TDG, I'd intended the Arcane Enforcer to represent what happens when a rogue mage ends their studies, as well as add an alternative to the very fragile Shadow Mage. But the main reason I included it in this PR is because I hadn't expected the sprites to need much work - it sounds like that's not the case, so I'll gladly remove them.

File: data/core/units/humans/Outlaw_Shadow_Lord.cfg (hunk 1 above)
Comment (Dalas121) [reply data/core/units/humans/Outlaw_Shadow_Lord.cfg:46]: Fixed, thanks

**Comment by doofus-01** [comment]
> In TDG, I'd intended the Arcane Enforcer to represent what happens when a Rogue Mage ends their studies, as well as add an alternative to the very fragile Shadow Mage. But the main reason I included it in this PR is because it's fully animated and I hadn't expected the sprites to need much work - it sounds like that's not the case, so I'll gladly remove them.

Just to be clear, I wasn't saying the Arcane Enforcer is a dumb idea with terrible artwork, I was just saying I didn't see a reason to merge it to core.  It's probably a sensible unit in context (and I hope to try out TDG someday soon).

**Comment by Dalas121** [comment]
Given all the turmoil going on with Wesnoth at the moment, I'm going to close this issue and instead work towards the much simpler option of only standardizing stats in mainline campaigns. This will also hopefully be ready sooner, which will help with their appearances in the upcoming TDG and TRoW changes.

**Comment by Gweddeoran** [comment]
@doofus-01 We have these really old portraits of the Rogue Mage line from BTIsaac [here](https://forums.wesnoth.org/viewtopic.php?t=47765). Do you think we can salvage these or have to work from scratch?

**Comment by doofus-01** [comment]
Editing someone else's work is often more work than starting from scratch.  It doesn't hurt to have the link there, but I'd say a fresh start is better.

**Pull request closed without merging**